net/miniupnpd/Makefile

```
@@ -8,12 +8,17 @@
 include $(TOPDIR)/rules.mk
 
 PKG_NAME:=miniupnpd
-PKG_VERSION:=2.3.7
+PKG_VERSION:=2.3.9
 PKG_RELEASE:=1
 
-PKG_SOURCE_URL:=https://miniupnp.tuxfamily.org/files
-PKG_SOURCE:=$(PKG_NAME)-$(PKG_VERSION).tar.gz
-PKG_HASH:=fbdd5501039730f04a8420ea2f8f54b7df63f9f04cde2dc67fa7371e80477bbe
+PKG_SOURCE_VERSION:=miniupnpd_$(shell echo $(PKG_VERSION) | sed 's/\./_/g')
+#PKG_SOURCE_VERSION:=d66872e34d9ff83a07f8b71371b13419b2089953
+PKG_SOURCE_URL:=https://codeload.github.com/miniupnp/miniupnp/tar.gz/$(PKG_SOURCE_VERSION)?
+PKG_SOURCE:=miniupnp-$(PKG_SOURCE_VERSION).tar.gz
+PKG_HASH:=ec7981351ad6a046eee0abf522ed1a45a3b0517e1da64e03826051f5f5354ea5
+PKG_BUILD_DIR:=$(BUILD_DIR)/$(PKG_NAME)-$(BUILD_VARIANT)/$(PKG_NAME)-$(PKG_VERSION)/miniupnpd
+TAR_OPTIONS+= --strip-components 1
+TAR_CMD=$(HOST_TAR) -C $(1)/.. $(TAR_OPTIONS)
```

This is getting more complicated than it should be. You are not using BUILD_VARIANT, thus it can be dropped.

---

```suggestion
PKG_NAME:=miniupnpd
PKG_VERSION:=2.3.9
PKG_RELEASE:=1

PKG_SOURCE_URL:=https://miniupnp.tuxfamily.org/files
PKG_SOURCE:=$(PKG_NAME)-$(PKG_VERSION).tar.gz
PKG_HASH:=66cb3c3d697ab2bb3a61d3c48628166d6ba328d7c2dbeb95898fdf2a3202af7b

PKG_MAINTAINER:=
PKG_LICENSE:=BSD-3-Clause
PKG_LICENSE_FILES:=LICENSE
PKG_CPE_ID:=cpe:/a:miniupnp_project:miniupnpd
```

This one works for me as it is. I haven't looked why your codeload does not work (yet). Maybe later.

---

I still dont understand. What issue do you have currently then it needs to be switched to codeload? The current solution works and there were two patches. Patches can be applied there and you do have some unknown issues, where it does not work. Patches still can be applied while downloading tarball from website.